distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -128,21 +137,16 @@ private void validateConfig(OptionSet options, Environment env) throws UserExcep
         }
     }
 
-    private static SecureString getKeystorePassword(KeyStoreWrapper keystore, Terminal terminal) {
-        if (keystore != null && keystore.hasPassword()) {
-            return new SecureString(terminal.readSecret(KeyStoreWrapper.PROMPT));
-        } else {
-            return new SecureString(new char[0]);
-        }
-    }
-
+    // Autoconfiguration of SecureSettings is currently only supported for KeyStore based secure settings
     private Environment autoConfigureSecurity(
         Terminal terminal,
         OptionSet options,
         ProcessInfo processInfo,
         Environment env,
-        SecureString keystorePassword
+        SecureString optionalPassword
```

Comment:
it's not really optional right? it's the same as it was before, it might be an empty string, but it still exists (not null, as optional might imply)